pkg/resourceinterpreter/defaultinterpreter/healthy.go

```diff
@@ -32,8 +32,16 @@ func interpretDeploymentHealth(object *unstructured.Unstructured) (bool, error)
 		return false, err
 	}
 
-	healthy := (deploy.Status.UpdatedReplicas == *deploy.Spec.Replicas) && (deploy.Generation == deploy.Status.ObservedGeneration)
-	return healthy, nil
+	if deploy.Status.UpdatedReplicas < *deploy.Spec.Replicas {
+		return false, nil
+	}
+	if deploy.Status.AvailableReplicas < deploy.Status.UpdatedReplicas {
+		return false, nil
+	}
+	if deploy.Generation != deploy.Status.ObservedGeneration {
+		return false, nil
+	}
```

Comment:
I think we can judge at first, the current object is up-to-date only when generation is consistent.